app/javascript/src/ActiveDocs/customize/wrappedComponents/Parameters.jsx

```diff
@@ -0,0 +1,114 @@
+  const isAutocompletable = customParamName in AUTOCOMPLETE_PARAMS_MAP
+  if (!isAutocompletable) {
+    return undefined
+  }
+  const paramValue = accountData[AUTOCOMPLETE_PARAMS_MAP[customParamName]]
+    .map(item => {
+      return {name: item.name, value: item.value}
+    })
+  return paramValue.length > 0 ? paramValue : undefined
+}
+
+const AutocompleteList = ({autocomopleteValues, isOpen, customParamName, onClickAutocomplete}) => {
+  return (
+    <tr className={`autocomplete-row-${customParamName}`}>
+      <td className="parameters-col_name"></td>
+      <td className="parameters-col_description">
+        <div className={`autocomplete-list autocomplete-list-${customParamName} ${isOpen ? 'show' : 'hide'}`}>
+          <div className='apidocs-param-tips'>
+            <ul>
+              { autocomopleteValues && autocomopleteValues.map(
+                (currentValue) => (
+                  <li className={`autocomplete-${customParamName}`} key={currentValue.name} data-value={`${currentValue.value}`} onClick={onClickAutocomplete}>
+                    <strong>{currentValue.name}</strong><br/>{currentValue.value}
+                  </li>
+                )
+              ) }
+            </ul>
+          </div>
+        </div>
+      </td>
+    </tr>
+  )
+}
+
+const CustomParameterWrapper = ({customParamName, customParamValue, children}) => {
+  const [isAutocompleteOpen, setIsAutocompleteOpen] = useState(false)
+
+  const customParamInput = document.querySelector(
+    `[data-param-name='${customParamName}'] .parameters-col_description input`
+  )
+
+  const handleClickAutocomplete = (e) => {
+    customParamInput.value = e.currentTarget.dataset.value
+    setIsAutocompleteOpen(false)
+  }
+
+  if (customParamInput) {
+    customParamInput.addEventListener('focus', () => {
+      setIsAutocompleteOpen(true)
+    })
+    document.body.addEventListener('click', (e) => {
+      if (e.target === customParamInput || e.target.classList.contains(`autocomplete-${customParamName}`)) {
+        return
+      }
+      setIsAutocompleteOpen(false)
+    })
+  }
+
+  return (
+    <>
+      {children}
+      <AutocompleteList
+        customParamName={customParamName}
+        autocomopleteValues={customParamValue}
+        isOpen={isAutocompleteOpen}
+        onClickAutocomplete={handleClickAutocomplete}
+        customParamName={customParamName}
+      />
+    </>
+  )
+}
+
+export const CustomParameterRow = (Original, system) => props => {
+  const accountData = system.accountData
+  const customParamName = props.rawParam._list._tail.array
+    .filter(param => param[0] === 'name')[0][1]
+  const customParamValue = getCustomParamValue(customParamName, accountData)
+
+  if (!customParamName || !customParamValue) {
+    return React.createElement(Original, props)
+  }
+
+  return React.createElement(
+    CustomParameterWrapper,
+    {customParamName, customParamValue},
+    React.createElement(Original, props)
+  )
+}
```

Comment:
I think there might be a better way to avoid the conditionals here, and in `getCustomParamValue`. 
the conditional maybe useful before you render `CustomParamenterRow`, then you'll know for sure that it can autocomplete. That will reduce your conditionals in these 2 fn to 1 outside.